packages/flutter_tools/lib/src/usage.dart

```diff
@@ -44,35 +44,43 @@ class Usage {
   Analytics _analytics;
 
   bool _printedUsage = false;
+  bool _suppressAnalytics = false;
```

Comment:
This is being used as a per-command suppression, something that would be an additional constraint over and above the opt-out. If they've opt-out, nothing is sent, and if they've suppressed analytics for this specific invocation, nothing is sent.

Generally, we expect people to opt-out or not; `--suppress-analytics` is meant to be used by machine driven systems, like driving some flutter processes from xcode.